Source/WebCore/html/HTMLTableElement.cpp

```diff
@@ -427,10 +427,8 @@ void HTMLTableElement::attributeChanged(const QualifiedName& name, const AtomStr
             m_rulesAttr = AllRules;
         break;
     case AttributeNames::cellpaddingAttr:
-        if (!newValue.isEmpty())
-            m_padding = std::max(0, parseHTMLInteger(newValue).value_or(0));
-        else
-            m_padding = 1;
+        if (auto cellPadding = parseHTMLInteger(newValue))
```

Comment:
```suggestion
        if (auto cellPadding = parseHTMLNonNegativeInteger(newValue))
```

This is probably not going to fix the tests, but I noticed this is more accurate according to the spec.

Reply:
Done.